jmix-flowui/flowui-kit/src/main/java/io/jmix/flowui/kit/component/codeeditor/JmixCodeEditor.java

```diff
@@ -0,0 +1,228 @@
+
+    protected CodeEditorValidationSupport validationSupport;
+
+    public JmixCodeEditor() {
+        super(PROPERTY_VALUE, "", true);
+
+        ComponentUtil.addListener(this, JmixCodeEditorValueChangedEvent.class, this::onCodeEditorValueChangedEvent);
+
+        initComponent();
+    }
+
+    protected void initComponent() {
+        if ((getElement().getProperty(PROPERTY_VALUE) == null)) {
+            setPresentationValue(getEmptyValue());
+        }
+
+        setInvalid(false);
+        addValueChangeListener(e -> validate());
+        addClientValidatedEventListener(e -> validate());
+    }
+
+    @Synchronize(PROPERTY_HIGHLIGHT_ACTIVE_LINE)
+    public boolean isHighlightActiveLine() {
+        return getElement().getProperty(PROPERTY_HIGHLIGHT_ACTIVE_LINE, true);
+    }
+
+    public void setHighlightActiveLine(boolean highlightActiveLine) {
+        getElement().setProperty(PROPERTY_HIGHLIGHT_ACTIVE_LINE, highlightActiveLine);
+    }
+
+    @Synchronize(PROPERTY_SHOW_GUTTER)
+    public boolean isShowGutter() {
+        return getElement().getProperty(PROPERTY_SHOW_GUTTER, true);
+    }
+
+    public void setShowGutter(boolean showGutter) {
+        getElement().setProperty(PROPERTY_SHOW_GUTTER, showGutter);
+    }
+
+    @Synchronize(PROPERTY_SHOW_LINE_NUMBERS)
+    public boolean isShowLineNumbers() {
+        return getElement().getProperty(PROPERTY_SHOW_LINE_NUMBERS, true);
+    }
+
+    public void setShowLineNumbers(boolean showLineNumbers) {
+        getElement().setProperty(PROPERTY_SHOW_LINE_NUMBERS, showLineNumbers);
+    }
+
+    @Synchronize(PROPERTY_SHOW_PRINT_MARGIN)
+    public boolean isShowPrintMargin() {
+        return getElement().getProperty(PROPERTY_SHOW_PRINT_MARGIN, true);
+    }
+
+    public void setShowPrintMargin(boolean showPrintMargin) {
+        getElement().setProperty(PROPERTY_SHOW_PRINT_MARGIN, showPrintMargin);
+    }
+
+    @Synchronize(PROPERTY_PRINT_MARGIN_COLUMN)
+    public int getPrintMarginColumn() {
+        return getElement().getProperty(PROPERTY_PRINT_MARGIN_COLUMN, 80);
+    }
+
+    public void setPrintMarginColumn(int printMarginColumn) {
+        getElement().setProperty(PROPERTY_PRINT_MARGIN_COLUMN, printMarginColumn);
+    }
+
+    @Synchronize(PROPERTY_THEME)
+    public CodeEditorTheme getTheme() {
+        return CodeEditorTheme.valueOf(getElement().getProperty(PROPERTY_THEME));
+    }
+
+    public void setTheme(CodeEditorTheme theme) {
+        getElement().setProperty(PROPERTY_THEME, theme.getThemeName());
+    }
+
+    @Synchronize(PROPERTY_MODE)
+    public CodeEditorMode getMode() {
+        return CodeEditorMode.valueOf(getElement().getProperty(PROPERTY_MODE));
+    }
+
+    public void setMode(CodeEditorMode mode) {
+        getElement().setProperty(PROPERTY_MODE, mode.getModeName());
+    }
+
+    @Synchronize(PROPERTY_FONT_SIZE)
+    public int getFontSize() {
+        return getElement().getProperty(PROPERTY_FONT_SIZE, 16);
```

Comment:
I don't like this magic number. Let's create a constant and set its value on component init.

Reply:
alternatively, return -1 as a default value as it shouldn't happen, because we have a default value on client side. But dob't forget a comment describing why we return -1